File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let map = new Map();
        let value = 0;

        for(let i = 0; i < nums.length; i++){
        if(!map.get(nums[i])){
            map.set(nums[i], 1);
        } else{
            value = map.get(nums[i]);    
            map.set(nums[i], value + 1);
        }
        }

        let arr = [...map];
        arr.sort((a,b) => b[1] - a[1]);

        let count = 0;
        let res = [];

        for(let [key, value] of arr){
            if(count === k) break;
            res.push(key);
            count++;
        }

        return res;
    }
}

File: Data Structures & Algorithms/subsets/submission-0.cs
public class Solution {
    public List<List<int>> Subsets(int[] nums) {
        var res = new List<List<int>>();
        res.Add(new List<int>());

        foreach (int num in nums){
            int resCount = res.Count;

            for(int i = 0; i < resCount; i++){
                var subset = new List<int>(res[i]);
                subset.Add(num);
                res.Add(subset);
            }
        }

        return res;
    }
}

/*
nums = [1,2,3]
res = [ [], [1], [2], [1,2], [3], [1,3], [2, 3], [1,2,3] ]
resCount = 4, i = 3

subset = [1,3]
num = 3
*/
